File: 프로그래머스/lv0/120882. 등수 매기기/등수 매기기.py
def solution(score):
    data = []
    for idx, s in enumerate(score):
        data.append([sum(s)/2, idx])
    
    data.sort(key=lambda x: -x[0])
    
    data[0].append(1)
    rank = 2
    for i in range(1, len(data)):
        if data[i-1][0] == data[i][0]:
            data[i].append(data[i-1][2])
        else:
            data[i].append(rank)
        rank += 1
    
    data.sort(key=lambda x: x[1])
    answer = []
    for d in data:
        answer.append(d[2])
    
    return answer
    

File: 프로그래머스/0/120882. 등수 매기기/등수 매기기.java
class Solution {
    public int[] solution(int[][] score) {
        double[] map = new double[score.length];
        int[] answer = new int[score.length];
        for (int i = 0; i < score.length; i++) {
            map[i] = (score[i][0] + score[i][1]) / 2.0;
        }

        for (int i = 0; i < score.length; i++) {
            int cnt = 0;
            for (int j = 0; j < score.length; j++) {
                if (map[i] < map[j]) {
                    cnt++;
                }
            }
            answer[i] = cnt + 1;
        }

        return answer;
    }
}